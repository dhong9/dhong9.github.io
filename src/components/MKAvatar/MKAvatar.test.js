/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

// React testing libraries
import { render } from "@testing-library/react";

// Material Kit 2 React themes
import { ThemeProvider } from "@mui/material/styles";
import theme from "assets/theme";

// Component to test
import MKAvatar from "components/MKAvatar";

describe("MKAvatar", () => {
  it("renders extra small transparent avatars", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKAvatar bgColor="transparent" size="xs"></MKAvatar>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders small primary avatars", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKAvatar bgColor="primary" size="sm"></MKAvatar>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders large secondary avatars", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKAvatar bgColor="secondary" size="lg"></MKAvatar>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders extra large info avatars", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKAvatar bgColor="info" size="xl"></MKAvatar>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders medium success avatars", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKAvatar bgColor="success" size="md"></MKAvatar>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
