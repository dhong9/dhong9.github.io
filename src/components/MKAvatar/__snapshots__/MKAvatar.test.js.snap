// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MKAvatar renders default success avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1b3h4cq-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit MuiAvatar-fallback css-1j1ftla-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders extra large info avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jbkl7d-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit MuiAvatar-fallback css-1j1ftla-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders extra small transparent avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1qwm04v-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit MuiAvatar-fallback css-1j1ftla-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders large secondary avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-kyvz8v-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit MuiAvatar-fallback css-1j1ftla-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders small primary avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-9c9o2o-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit MuiAvatar-fallback css-1j1ftla-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;
